server/app/auth/oidc/CustomOidcLogoutRequest.java

```diff
@@ -0,0 +1,103 @@
+package auth.oidc;
+
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableMap;
+import com.nimbusds.openid.connect.sdk.LogoutRequest;
+import java.net.URI;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Custom Logout Request that allows for divergence from the [oidc
+ * spec](https://openid.net/specs/openid-connect-rpinitiated-1_0.html) if your provider requires it
+ * (e.g. Auth0).
+ *
+ * <p>Does not provide the recommended id_token_hint, since these are not stored by the civiform
+ * profile.
+ *
+ * <p>Uses the post_logout_redirect_uri parameter by default, but allows overriding to a different
+ * value
+ *
+ * <p>Allows adding extra custom query parameters to the URL.
+ */
+public final class CustomOidcLogoutRequest extends LogoutRequest {
+  /** The optional post-logout redirection query param. */
+  private final String postLogoutRedirectParam;
+
+  /** The optional post-logout redirection URI. */
+  private final URI postLogoutRedirectURI;
+
+  /** Pptional extra query params to add to the URL. */
+  private final ImmutableMap<String, String> extraParams;
+
+  /**
+   * Create new OIDC logout request with a optional redirect url, optional client id, and other
+   * params.
+   */
+  public CustomOidcLogoutRequest(
+      final URI uri,
+      final String postLogoutRedirectParam,
+      final URI postLogoutRedirectURI,
+      final ImmutableMap<String, String> extraParams) {
+
+    super(uri, /* idTokenHint */ null, postLogoutRedirectURI, /* state */ null);
```

Comment:
Does this work? The base class constructor appears to throw an IllegalArgumentException when the provided hint is null.

Reply:
I'm not sure where you see that?  I see this in the docs:
```
idTokenHint - The ID token hint (recommended), null if not specified.
```
https://www.javadoc.io/doc/com.nimbusds/oauth2-oidc-sdk/5.25/com/nimbusds/openid/connect/sdk/LogoutRequest.html

Reply:
Here's where I found it: https://github.com/hidglobal/oauth-2.0-sdk-with-openid-connect-extensions/blob/master/src/main/java/com/nimbusds/openid/connect/sdk/LogoutRequest.java#L90

I think I might have been looking at the wrong implementation.